src/sentry/auth/helper.py

```diff
@@ -101,15 +102,32 @@ class AuthIdentityHandler:
     identity: Mapping[str, Any]
 
     def __post_init__(self) -> None:
-        self.user: Union[User, AnonymousUser] = self._initialize_user()
+        self._user = None
+
+    @property
+    def user(self) -> Union[User, AnonymousUser]:
+        if self._user is not None:
+            return self._user
 
-    def _initialize_user(self) -> Union[User, AnonymousUser]:
         email = self.identity.get("email")
         if email:
-            user = resolve_email_to_user(email)
-            if user:
-                return user
-        return self.request.user
+            try:
+                self._user = resolve_email_to_user(email)
+            except AmbiguousUserFromEmail as e:
+                self._user = e.users[0]
+                self.warn_about_ambiguous_email(email, e.users, self._user)
+        if self._user is None:
+            self._user = self.request.user
+        return self._user
+
+    @staticmethod
+    def warn_about_ambiguous_email(email: str, users: Tuple[User], chosen_user: User):
+        with sentry_sdk.push_scope() as scope:
+            scope.level = "warning"
+            scope.set_tag("email", email)
+            scope.set_extra("user_ids", [user.id for user in users])
+            scope.set_extra("chosen_user", chosen_user.id)
+            sentry_sdk.capture_message("Handling identity from ambiguous email address")
```

Comment:
Sentry dogfood question: now that there is an exception object, is `capture_exception` better than `capture_message` here? (I'm thinking maybe not, as we're purposefully swallowing the exception.)